src/skin/qml/asyncimageprovider.cpp

```diff
@@ -0,0 +1,85 @@
+#include "skin/qml/asyncimageprovider.h"
+
+#include "library/coverartcache.h"
+#include "track/track.h"
+
+namespace {
+const QString kCoverArtPrefix = QStringLiteral("coverart/");
+}
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+
+AsyncImageResponse::AsyncImageResponse(
+        std::shared_ptr<TrackCollectionManager> pTrackCollectionManager,
+        const QString& id,
+        const QSize& requestedSize)
+        : m_pTrackCollectionManager(pTrackCollectionManager),
+          m_id(id),
+          m_requestedSize(requestedSize) {
+    setAutoDelete(false);
+}
+
+QQuickTextureFactory* AsyncImageResponse::textureFactory() const {
+    return QQuickTextureFactory::textureFactoryForImage(m_image);
+}
+
+void AsyncImageResponse::run() {
+    if (m_id.startsWith(kCoverArtPrefix)) {
+        QString trackLocation = AsyncImageProvider::coverArtUrlIdToTrackLocation(m_id);
+
+        TrackPointer pTrack = m_pTrackCollectionManager->getTrackByRef(
+                TrackRef::fromFilePath(trackLocation));
+
+        if (!pTrack) {
+            qWarning() << "ImageProvider: Track with location" << trackLocation << "not found";
+            return;
+        }
+
+        CoverInfo coverInfo = pTrack->getCoverInfoWithLocation();
+        CoverInfo::LoadedImage loadedImage = coverInfo.loadImage();
+        if (loadedImage.result != CoverInfo::LoadedImage::Result::Ok) {
```

Comment:
This fallback is wrong. If CoverInfo cannot load the image we should not try to overrule this logic.